travis/getaddons.py

```diff
@@ -84,7 +84,138 @@ def get_addons(path):
         res = [os.path.join(path, x)
                for x in os.listdir(path)
                if is_addons(os.path.join(path, x))]
-    return res
+    deps = dict([(addon, get_dependencies(addon))
+                 for addon in res])
+    sorted_addons = get_sorted_addons_by_level(deps)
+    return sorted_addons
+
+
+def get_dependencies(path):
+    """Gets the dependencies of an addon reading the `oca_dependencies.txt`
+    file.
+
+    :param path: The addon path
+    :rtype: str
+
+    :returns: The dependency list
+    :rtype: list
+    """
+    deps = []
+    dep_files = ['oca_dependencies.txt']
+    for dep_file in dep_files:
+        dep_path = os.path.join(path, dep_file)
+        new_deps = []
+        try:
+            with open(dep_path) as dep:
+                for line in dep:
+                    line = line.strip()
+                    if not line or line.startswith('#'):
+                        continue
+                    new_deps.append(line.split()[0])
+        except IOError:
+            continue
+        deps = deps + new_deps
```

Comment:
Why don't you use only one list?

Reply:
I should have. Changed it to use one list and append directly to that one.